pkg/monitortests/node/watchnodes/monitortest.go

```diff
@@ -53,3 +59,57 @@ func (*nodeWatcher) Cleanup(ctx context.Context) error {
 	// TODO wire up the start to a context we can kill here
 	return nil
 }
+
+func unexpectedNodeNotReadyJunit(finalIntervals monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	const testName = "[sig-node] node-lifecycle detects unexpected not ready node"
+	var failures []string
+	for _, event := range finalIntervals {
+		if strings.Contains(event.Message.HumanMessage, "unexpected node not ready") {
+			failures = append(failures, fmt.Sprintf("%v - %v", event.Locator.OldLocator(), event.Message.OldMessage()))
+		}
+	}
+
+	// failures during a run always fail the test suite
+	var tests []*junitapi.JUnitTestCase
+	if len(failures) > 0 {
+		tests = append(tests, &junitapi.JUnitTestCase{
+			Name:      testName,
+			SystemOut: strings.Join(failures, "\n"),
+			FailureOutput: &junitapi.FailureOutput{
+				Output: fmt.Sprintf("node-lifecycle reports %d unexpected notReady events. The node went NotReady in an unpredicted way.\n\n%v", len(failures), strings.Join(failures, "\n")),
+			},
+		})
+	}
+
+	if len(tests) == 0 {
+		tests = append(tests, &junitapi.JUnitTestCase{Name: testName})
+	}
+	return tests
+}
+
+func unreachableNodeTaint(finalIntervals monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	const testName = "[sig-node] node-lifecycle detects unreachable state on node"
+	var failures []string
+	for _, event := range finalIntervals {
+		if strings.Contains(event.Message.HumanMessage, "unexpected node unreachable") {
+			failures = append(failures, fmt.Sprintf("%v - %v", event.Locator.OldLocator(), event.Message.OldMessage()))
```

Comment:
I would output the time the problem occurred in the test output as well, the info is here on event.

Reply:
So this is the error so the test fails. The interval should display the time.

In this case, I want an error if I detect any intervals that match this reason. So it would be multiple times in this message which may be confusing.

Reply:
Ok that's fine but I would include event.From and .To, test output should be clear about when the problem was detected.

Reply:
Probably a better idea to check event.Reason == NodeUnreachableReason as well, that is meant for programatic comparison and uses a constant.

Reply:
I'll address this after the CI reports these changes. Just want to see if this works right now.

Reply:
> Probably a better idea to check event.Reason == NodeUnreachableReason as well, that is meant for programatic comparison and uses a constant.

Good call. I have that stashed once I see some jobs run.

Reply:
pushed. PTAL.